Service/AuditTrailService.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+// src/Service/AuditTrailService.php
+
+namespace CommonGateway\ZGWBundle\Service;
+
+use App\Entity\Endpoint;
+use App\Entity\File;
+use App\Entity\ObjectEntity;
+use CommonGateway\CoreBundle\Service\GatewayResourceService;
+use CommonGateway\CoreBundle\Service\MappingService;
+use Doctrine\ORM\EntityManagerInterface;
+use Ramsey\Uuid\Uuid;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+use CommonGateway\CoreBundle\Service\CacheService;
+
+class AuditTrailService
+{
+    private array $configuration;
+    private array $data;
+
+    private EntityManagerInterface $entityManager;
+    private ParameterBagInterface $parameterBag;
+    private CacheService $cacheService;
+    private MappingService $mappingService;
+    private GatewayResourceService $resourceService;
+
+    public function __construct(
+        EntityManagerInterface $entityManager,
+        ParameterBagInterface $parameterBag,
+        CacheService $cacheService,
+        MappingService $mappingService,
+        GatewayResourceService $resourceService
+    ){
+        $this->entityManager = $entityManager;
+        $this->parameterBag = $parameterBag;
+        $this->cacheService = $cacheService;
+        $this->mappingService = $mappingService;
+        $this->resourceService = $resourceService;
+    }
+
+    /**
+     * Returns the audit trails of the given object
+     *
+     * @param array $data the data of the request
+     * @param array $configuration the configuration of the request
+     *
+     * @return array the audit trails of the object
+     */
+    public function auditTrailHandler(array $data, array $configuration): array
+    {
+        $this->data = $data;
+        $this->configuration = $configuration;
+
+        if (key_exists('{audit trail._self.id}', $this->data['path']) === true) {
+            $objectId = $this->data['path']['{audit trail._self.id}'];
+        }
+
+        if (key_exists('id', $this->data['path']) === true) {
```

Comment:
This feels like it should be an else if somehow because both if statements set $objectId